.github/workflows/ci.yml

```diff
@@ -200,22 +262,45 @@ jobs:
         with:
           path: |
             ./target
-          key: |
-            s0mth1ng_rand0m # Cache never be created for this key.
+            !./target/gn_out
+            !./target/*/*.zip
+            !./target/*/*.tar.gz
+          key: never_saved
           restore-keys: |
-            a-cargo-target-${{ matrix.os }}-${{ matrix.profile }}-${{ hashFiles('Cargo.lock') }}-
+            z-cargo-target-${{ matrix.os }}-${{ matrix.profile }}-
 
-      # Skips saving cache in PR branches
+      # Don't save cache after building PRs or branches other than 'main'.
       - name: Skip save cache (PR)
         run: echo "CACHE_SKIP_SAVE=true" >> $GITHUB_ENV
         shell: bash
         if: github.ref != 'refs/heads/main'
 
       - name: Apply and update mtime cache
+        if: matrix.profile == 'release'
         uses: ./.github/mtime_cache
         with:
           cache-path: ./target
 
+      - name: Shallow clone crates.io index
+        shell: bash
+        run: |
+          if [ ! -d ~/.cargo/registry/index/github.com-1ecc6299db9ec823/.git ]
+          then
+            git clone --depth 1 --no-checkout                      \
+                      https://github.com/rust-lang/crates.io-index \
+                      ~/.cargo/registry/index/github.com-1ecc6299db9ec823
+          fi
```

Comment:
I don't understand this. Is this related to the sysroot?

Reply:
Separated out the commit

Reply:
What does this do tho? What is "1ecc6299db9ec823" ?

Reply:
> What does this do tho?

As reflected in the name, it makes a shallow clone of the crates.io index.

> What is "1ecc6299db9ec823"

I don't know.

> Will it ever need to be updated?

I don't know. But in my Rust-using career it has never changed.

> I think this could have a comment as to the purpose

The purpose is to make CI faster. Want me add `# This makes CI faster`?

Reply:
```
# Manually clone the cargo crate index shallowly so that cargo doesn't
# clone it deeply. This saves about 1 minute of CI time. We don't believe the
# identifier '1ecc6299db9ec823' will ever change, but if it does then this command
# must be updated.
```